specs/UpdateRuntime.md

```diff
@@ -0,0 +1,229 @@
+# Background
+The new version of an app might require a newer version of Edge WebView2 Runtime.
+
+An app updater may wish to ensure that a particular minimum version of the Edge WebView2 Runtime is installed before upgrading the app to a version that requires those features.
+
+Alternatively, the newer version of the app could use feature detection to disable portions of the app that rely on new WebView2 Runtime features. In this alternate scenario, the app updater would install the new version of the app immediately, and then request that the Edge WebView2 Runtime be updated, so that the updated app can start taking advantage of the new features.
+
+Edge WebView2 Runtime is auto updated and normally the latest version should be already installed. However, there could be cases that we need trigger Edge WebView2 Runtime
+update to ensure coordinated app and WebView2 Runtime update.
+
+# Description
+You may call the `UpdateRuntime` API to check and install updates to installed Edge WebView2 Runtime. This is useful when the app wants to coordinate app and
+Edge WebView2 Runtime update.
+
+# Examples
+## .NET, WinRT
+```c#
+using Microsoft.Web.WebView2.Core;
+async protected bool EnsureWebView2RuntimeVersion(string minimalVersionRequired)
+{
+    string currentRuntimeVersion = CoreWebView2Environment.GetAvailableBrowserVersionString();
+    if (CoreWebView2Environment.CompareBrowserVersions(currentRuntimeVersion, minimalVersionRequired) < 0)
+    {
+      auto environment = await CoreWebView2Environment.CreateAsync();
+      auto updateResult = await environment.UpdateRuntimeAsync();
+      if (updateResult.UpdateRuntimeStatus != CoreWebView2RuntimeUpdateStatus.LatestVersionInstalled)
+        return false;
+      // check runtime version again
+      currentRuntimeVersion = CoreWebView2Environment.GetAvailableBrowserVersionString();
+    }
+    return (CoreWebView2Environment.CompareBrowserVersions(currentRuntimeVersion, minimalVersionRequired) >= 0);
+}
+
+// For the scenario where the app wants to light up features fast while running with the old version.
+{
+    ...
+    // Listen to NewBrowserVersionAvailable to take action.
+    webView2Environment.NewBrowserVersionAvailable += delegate (object sender, object args)
+    {
+        // See the NewBrowserVersionAvailable documentation for more information
+        // Close current WebView2 Control
+        // Wait for it to completely shutdown
+        // Recreate WebView2 Control to run with newer version
+    };
+    
+    // Trigger Edge WebView2 Runtime update, ignore update result and rely on NewBrowserVersionAvailable to take action.
+    EnsureWebView2RuntimeVersion(desiredVersion);
```

Comment:
Maybe just a syntax nit to be use a throwaway: `_ = EnsureWeb...`.

But what I was really expecting here was an await call, and unless/until it returned 'true' the app would disable portions of itself for the features not available.

Reply:
Show two scenarios as two samples with the one helper one for min version and one for light up. For light up, make it clearer with something like awaiting the Ensure method and then if it succeeds running a 'StartLightupFeatures' method or something like that.

Reply:
The light up actually would only happen after recreate the webview/restart the app. I can add some code/comment showing how to do feature detection before using a feature.
Waiting on ensure method itself would not work, as ensure could also mean that we are already running latest version, and the app should move to newer version to light up even if the update is installed automatically. So, listening to new version available event is a better way to switch.

Reply:
Updated the sample code and added the need to check currently running version for whether to light up new features.